### ISSUE TITLE
xr.where not picking up nan from another NetCDF file

### ISSUE THREAD

**Issue by raybellwaves** [pydata/xarray#1381]
I posted this question here:
http://stackoverflow.com/questions/43485347/python-xarray-copy-nan-from-one-dataarray-to-another

I have uploaded the files and code to https://github.com/raybellwaves/xarray_issue.git

When I originally discovered the issue I was using python version 
3.6.0 (default, Jan 28 2017, 13:49:14) 
[GCC Intel(R) C++ gcc 4.4 mode]

I tested the code with version
2.7.13 | packaged by conda-forge | (default, Mar 20 2017, 14:26:36) 
[GCC 4.2.1 Compatible Apple LLVM 6.1.0 (clang-602.0.53)]
and I did not get the error message which I posted in the question.
The code worked by the variable ws10_masked is empty

**Comment by shoyer** [comment]
@raybellwaves Thanks for following up with details!

Can you please share the versions of xarray and netCDF4-python you were using when you encountered this issue?

(I have not been able to reproduce this yet)

**Comment by raybellwaves** [comment]
@shoyer thanks for looking into this. 

Firstly, I updated the hs netCDF file as it was a different size then expected: (1, **82**, 131) instead of (1, **81**, 181). I've corrected this now but I am still getting an error message.

I've added the version numbers and outputs to https://github.com/raybellwaves/xarray_issue/blob/master/xr_where_issue.py

I have two setups. I run my model experiments and do most of my analysis on a HPC. However, I do most of my plotting on my laptop for ease of viewing. To help me debug this i've run the code on both setups (which each have different python version etc.) but a priority is the more up to date HPC python version. I've including the output from both though.
